packages/modules/order/src/utils/transform-order.ts

```diff
@@ -29,30 +43,118 @@ export function formatOrder(
       }
     })
 
-    order.shipping_methods = order.shipping_methods?.map((shippingMethod) => {
-      const sm = { ...shippingMethod.shipping_method }
+    if (isRelatedEntity) {
+      if (order.return) {
+        formatOrderReturn(order.return, mainOrder)
+      }
 
-      delete shippingMethod.shipping_method
-      return {
-        ...sm,
-        order_id: shippingMethod.order_id,
-        detail: {
-          ...shippingMethod,
-        },
+      if (options.entity === OrderClaim) {
+        formatClaim(order)
+      } else if (options.entity === OrderExchange) {
+        formatExchange(order)
+      } else if (options.entity === Return) {
+        formatReturn(order)
       }
-    })
+    }
+
+    if (order.shipping_methods) {
+      order.shipping_methods = order.shipping_methods?.map((shippingMethod) => {
+        const sm = { ...shippingMethod.shipping_method }
+
+        delete shippingMethod.shipping_method
+        return {
+          ...sm,
+          order_id: shippingMethod.order_id,
+          detail: {
+            ...shippingMethod,
+          },
+        }
+      })
+    }
 
-    order.summary = order.summary?.[0]?.totals
+    if (mainOrder.summary) {
+      mainOrder.summary = mainOrder.summary?.[0]?.totals
+    }
 
-    return options?.includeTotals
-      ? createRawPropertiesFromBigNumber(decorateCartTotals(order))
-      : order
+    return createRawPropertiesFromBigNumber(
+      options?.includeTotals ? decorateCartTotals(order) : order
+    )
   })
 
   return isArray ? orders : orders[0]
 }
 
-export function mapRepositoryToOrderModel(config) {
+function formatOrderReturn(orderReturn, mainOrder) {
+  orderReturn.items = orderReturn.items.filter(
+    (item) => !item.is_additional_item
+  )
+  orderReturn.items.forEach((orderItem) => {
+    const item = mainOrder.items?.find((item) => item.id === orderItem.item_id)
+
+    orderItem.detail = item?.detail
+  })
+}
+
+function formatClaim(claim) {
+  if (claim.additional_items) {
+    claim.additional_items = claim.additional_items.filter(
+      (item) => item.is_additional_item
+    )
+
+    claim.additional_items.forEach((orderItem) => {
+      const item = claim.order.items?.find(
+        (item) => item.id === orderItem.item_id
+      )
+
+      orderItem.detail = item?.detail
+    })
```

Comment:
question: I am not sure I understand this part. 

First, we filter away `additional_items` for items that are not additional – what items are these then?

Then, we for each additional item, we retrieve its corresponding item on the claim order – wouldn't these be new items and therefore not be part of the order (yet)?

Reply:
Claimed items and additional items, both are in the same table and just reference to the line item of the order.
Here we just filter and split them by the claimed items and the additional ones.
That filter could be at the SQL level as well, but that would make it more complex.

Reply:
Got it. I think what seems strange to me is that this piece of code tells me that `claim.additional_items` can have items that are not additional items and have to be filtered out. I am probably just missing some context here, but thought I'd share.

On line 100-102:
```
claim.additional_items = claim.additional_items.filter(
  (item) => item.is_additional_item
)
```